runtime/src/main/java/org/corfudb/util/Sleep.java

```diff
@@ -0,0 +1,120 @@
+package org.corfudb.util;
+
+import java.util.concurrent.TimeUnit;
+
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuInterruptedError;
+
+public enum Sleep {
+    NANOSECONDS {
+        @Override
+        long toMillis(long duration) {
+            return TimeUnit.NANOSECONDS.toMillis(duration);
+        }
+
+        @Override
+        int excessNanos(long duration, long milliseconds) {
+            return (int)(duration - (milliseconds * (1000L * 1000L)));
+        }
+    },
+    MICROSECONDS {
+        @Override
+        long toMillis(long duration) {
+            return TimeUnit.MICROSECONDS.toMillis(duration);
+        }
+
+        @Override
+        int excessNanos(long duration, long milliseconds) {
+            return (int)((duration * 1000L) - (milliseconds * (1000L * 1000L)));
+        }
+    },
+    MILLISECONDS {
+        @Override
+        long toMillis(long duration) {
+            return TimeUnit.MILLISECONDS.toMillis(duration);
+        }
+    },
+    SECONDS {
+        @Override
+        long toMillis(long duration) {
+            return TimeUnit.SECONDS.toMillis(duration);
+        }
+    },
+    MINUTES {
+        @Override
+        long toMillis(long duration) {
+            return TimeUnit.MINUTES.toMillis(duration);
+        }
+    },
+    HOURS {
+        @Override
+        long toMillis(long duration) {
+            return TimeUnit.HOURS.toMillis(duration);
+        }
+    },
+    DAYS {
+        @Override
+        long toMillis(long duration) {
+            return TimeUnit.DAYS.toMillis(duration);
+        }
+    };
+
+    /** Convert this duration to millis. Used for the millisecond parameter to sleep.
+     *
+     * @param duration  The duration to convert.
+     * @return          The duration in milliseconds.
+     */
+    abstract long toMillis(long duration);
+
+
+    /** Convert this duration to the excess nanoseconds parameter used for sleep. Only
+     *  applicable for units less than a millisecond.
+     * @param duration      The duration to convert.
+     * @param milliseconds  The amount of milliseconds.
+     * @return              The number of nanoseconds leftover by subtracting milliseconds
+     *                      from duration.
+     */
+    int excessNanos(long duration, long milliseconds) {
+        return 0;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this method and declare a constant for this value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3400)